core/src/test/java/org/springframework/security/provisioning/JdbcUserDetailsManagerTests.java

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2002-2022 the original author or authors.
+ * Copyright 2002-2023 the original author or authors.
```

Comment:
Gotcha.

I did it this way so any tests that were written for the class-to-be-deprecated would make it into the test class for the newer version once the class was deleted. Same thing goes for changes to the behaviour in the class to be tested.

Have you got some process I should be aware of that handles the same concern - getting all functionality consistently migrated over? I am not asking to be an arse, but to avoid extra work for us both if I trip over an unseen team practice lurking just under the horizon.

Reply:
Good question. We usually copy the tests over to a new file that matches the naming convention for the new class. That works since we also have a policy of not adding features to deprecated classes.